### DISCUSSION TITLE
Programmatically handle open/close of Menu component similar to Dialog

### DISCUSSION THREAD

**Discussion by Fabioni** [tailwindlabs/headlessui#2974] [Ideas]
I have a case where I want to manually handle when the menu open and closes without `<Menu.Button>` but also do not want to use a `<Dialog>` since I want the keyboard-arrow feature.

**Comment by atolz** [comment]
I have a similar issue, but I also want to be able to listen for events on the Menu, like when the menu is closed...

Eg.` <Menu onClose={closeAction}/>`

This way I can trigger actions when the menu closes.
